invent-stock-app/main.go

```diff
@@ -1,7 +1,223 @@
 package main
 
-import "fmt"
+import (
+	"bufio"
+	"database/sql"
+	"fmt"
+	"log"
+	"net/mail"
+	"os"
+	"strconv"
+	"strings"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/joho/godotenv"
+)
+
+type Product struct {
+	Name string
+	Price float64
+	Stock int
+}
+
+type Staff struct {
+	Name string
+	Email string
+	Position string
+}
+
+func displayMenu () {
+
+	menu := map[int]string{
+		1: "Add Product",
+		2: "Change Stock",
+		3: "Add Staff", 
+		4: "Generate Sales Report",
+		5: "Exit",
+	}
+	fmt.Println("==========================================")
+	fmt.Println("Welcome to Store Management CLI Interface")
+	fmt.Println("==========================================")
+	fmt.Println("")
+	fmt.Println("Please select menu from below...")
+	fmt.Println("")
+	for i , v := range menu{
+		fmt.Printf("%d. %s\n", i, v)
+	}
+}
 
 func main() {
-	fmt.Println("Semangat...")
+	//load .env file
+	err := godotenv.Load("../.env")
+	if err != nil {
+	  log.Fatal("Error loading .env file ", err)
+	}
+
+	//db connection
+	dns := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", 
+		os.Getenv("DB_USERNAME"),
+		os.Getenv("DB_PASSWORD"),
+		os.Getenv("DB_HOST"),
+		os.Getenv("DB_PORT"),
+		os.Getenv("DB_NAME"),
+	)
+	db, err := sql.Open("mysql", dns)
+	if err != nil {
+		log.Fatal("Error connecting to database ", err)
+	}
+	defer db.Close()
+
+	fmt.Println("Connection to db successful ")
+	fmt.Println("")
+
+	//init bufio reader
+	reader := bufio.NewReader(os.Stdout)
+
+	//run apps
+	menuLoop: for{
+		displayMenu()
+		fmt.Printf("Answer (1/2/3/4/5): ")
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatal("Error reading input ", err)
+		}
+		input = strings.TrimSpace(input)
+		switch input {
+			case "1":
+				product := getAddProdParam(reader)
+				addProduct(db, product.Name, product.Price, product.Stock)
+				fmt.Println("")
+			case "2":
+				break menuLoop;
+			case "3":
+				staff := getAddStaffParam(reader)
+				addStaff(db, staff.Name, staff.Email, staff.Position)
+				fmt.Println("")
+			case "4":
+				break menuLoop;
+			case "5":
+				break menuLoop;
+		}
+	}
+}
+func isValidEmail(email string) bool {
+	_, err := mail.ParseAddress(email)
+	return err == nil
 }
+
+func getAddStaffParam(reader *bufio.Reader) Staff {
+	//get product name 
+	fmt.Printf("Please insert staff name: ")
+	staffName, err := reader.ReadString('\n')
+	if err != nil {
+		log.Fatal("Error reading staff name input ", err)
+	}
+	staffName = strings.TrimSpace(staffName)
+	if staffName == "" {
+		log.Println("Staff name should not be empty")
+		return getAddStaffParam(reader)
```

Comment:
maybe it will be better if we just return nil? so that the user doesn't get stuck in input loop and can exit if they want to

Reply:
hmm, apa baiknya kita bikin another function yang bisa milih apa user mau reinput atau exit?

func choose () {

input, err := reader.ReadStr()
....

switch input{
case reinput:
//recall the input function
}
case exit:
//return to main menu

}